lib/seek/samples/sample_type_editing_constraints.rb

```diff
@@ -49,24 +49,17 @@ def allow_attribute_removal?(attr)
         end
       end
 
-      # whether a new attribtue can be created
-      def allow_new_attribute?
-        !samples?
-      end
-
       # whether the name of the attribute can be changed
       def allow_name_change?(attr)
         if attr.is_a?(SampleAttribute)
           return true if attr.new_record?
-          return false if inherited?(attr)
 
           attr = attr.accessor_name
         end
-        if attr
-          !samples?
-        else
-          true
-        end
+        return !samples? unless attr
```

Comment:
Is this logic correct? Previously it would allow name changes if attr was nil

Reply:
If it has to return `true` anyway and we don't care about sample permissions, I guess this method can go because it would always return `true`.